core/src/main/java/org/jdbi/v3/core/mapper/EnumMapper.java

```diff
@@ -50,50 +60,61 @@
      * @return an enum mapper that matches on {@link Enum#ordinal()}
      */
     public static <E extends Enum<E>> ColumnMapper<E> byOrdinal(Class<E> type) {
-        return new ByOrdinal<>(type);
+        return new EnumByOrdinalColumnMapper<>(type);
     }
 
-    private static class ByName<E extends Enum<E>> extends EnumMapper<E> {
-        private final Class<E> type;
-        private final ConcurrentMap<String, E> insensitiveLookup = new ConcurrentHashMap<>();
+    static class EnumByNameColumnMapper<E extends Enum<E>> implements ColumnMapper<E> {
+        private final Map<String, E> nameValueCache = new ConcurrentHashMap<>();
+        private final Class<E> enumClass;
 
-        private ByName(Class<E> type) {
-            this.type = type;
+        private EnumByNameColumnMapper(Class<E> enumClass) {
+            this.enumClass = enumClass;
         }
 
         @Override
-        public E map(ResultSet r, int columnNumber, StatementContext ctx) throws SQLException {
-            String name = r.getString(columnNumber);
-            return name == null ? null : insensitiveLookup.computeIfAbsent(name, this::resolve);
+        public E map(ResultSet rs, int columnNumber, StatementContext ctx) throws SQLException {
+            String name = ctx.findColumnMapperFor(String.class)
+                .orElseThrow(() -> new UnableToProduceResultException("a String column mapper is required to map Enums from names", ctx))
+                .map(rs, columnNumber, ctx);
+
+            return name == null ? null : nameValueCache.computeIfAbsent(name.toLowerCase(), lowercased -> getValueByName(enumClass, name, ctx));
```

Comment:
I wonder if the cost of lowercasing outweighs the cost of duplicate entries.  Previously I'd operated under the assumption that most users, even if the casing is wrong, will tend to be at least consistently wrong - in which case you'd have few if any truly redundant entries.  And even if they aren't, the extra cache entries may well be cheaper than repeatedly lowercasing.  Part of the reason for adding this cache in the first place is to avoid case sensitive operations.
I worry that this may be a performance regression if that is indeed the common case.

Reply:
I tried it with and without `toLowerCase()` in benchmarks, and it was way worse. I took it out.

Reply:
Thanks for checking!  String operations are surprisingly expensive.